### PR TITLE
opcode1negate push -1 on to the stack

### PR DESCRIPTION
**Pull request by mubarak23**
- [x] issue #3 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

Implemented code for opcode 1 negate that number -1 is pushed onto the stack.

### CODE REVIEW

**Comment by mubarak23** [comment]
The test is failing because of the below error. 
 
String literals can only include ASCII characters. on this

let expected_stack = array!["\x81"];

\x81 is a hexadecimal representation of a character with the code 129, which is not an ASCII character.

File: src/opcodes/tests/test_opcodes.cairo [hunk 1]
```diff
@@ -691,3 +691,21 @@ fn test_op_lessthan_equal() {
...
+    let bytecode = compiler.compile(program);
+    let mut engine = EngineTraitImpl::new(bytecode);
+    // let _ = engine.step();
+    let res = engine.step();
+    assert!(res, "Execution of run failed");
+    let dstack = engine.get_dstack();
+
+    assert_eq!(dstack.len(), 1, "Stack length is not 1");
+
+    let expected_stack = array!["\x81"];
```
Comment (j1mbo64) [inline src/opcodes/tests/test_opcodes.cairo:708]: Hello, maybe you can try to construct the `expected_stack` like this :
```
let mut expected_bytes: ByteArray = Default::default();
expected_bytes.append_byte(129);
let mut expected_stack = ArrayTrait::<ByteArray>::new();
expected_stack.append(expected_bytes);
assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
```

**Review by j1mbo64 — CHANGES_REQUESTED** [review]

File: src/opcodes/tests/test_opcodes.cairo (hunk 1 above)
Comment (j1mbo64) [reply src/opcodes/tests/test_opcodes.cairo:708]: This is way better sorry `let expected_stack = array![129];`

Comment (b-j-roberts) [reply src/opcodes/tests/test_opcodes.cairo:708]: There is a utility function for this, you can use `utils::int_to_bytes(-1)`, so something like `let expected_stack = array![utils::int_to_bytes(-1)]`

**Pull request merged**